presto-mysql/src/main/java/com/facebook/presto/plugin/mysql/MySqlClient.java

```diff
@@ -196,6 +219,66 @@ protected String toSqlType(Type type)
         return super.toSqlType(type);
     }
 
+    @Override
+    public PreparedStatement buildSql(ConnectorSession session, Connection connection, JdbcSplit split, List<JdbcColumnHandle> columnHandles) throws SQLException
+    {
+        Map<String, String> columnExpressions = columnHandles.stream()
+                .filter(handle -> handle.getJdbcTypeHandle().getJdbcTypeName().equalsIgnoreCase(GEOMETRY))
+                .map(handle -> handle.getColumnName())
+                .collect(toImmutableMap(
+                        identity(),
+                        columnName -> "ST_AsBinary(`" + columnName + "`)"));
+
+        return new QueryBuilder(identifierQuote).buildSql(
+                this,
+                session,
+                connection,
+                split.getCatalogName(),
+                split.getSchemaName(),
+                split.getTableName(),
+                columnHandles,
+                columnExpressions,
+                split.getTupleDomain(),
+                split.getAdditionalPredicate());
+    }
+
+    @Override
+    public Optional<ReadMapping> toPrestoType(ConnectorSession session, JdbcTypeHandle typeHandle)
+    {
+        String typeName = typeHandle.getJdbcTypeName();
+
+        if (typeName.equalsIgnoreCase(GEOMETRY)) {
+            return Optional.of(geometryReadMapping());
+        }
+
+        return super.toPrestoType(session, typeHandle);
+    }
+
+    private ReadMapping geometryReadMapping()
+    {
+        return sliceReadMapping(VARCHAR,
```

Comment:
Let's also write a unit test for this which supplied different valid and invalid `resultSet`s to the read mapping function and verify that it is properly converted. We can create a new file `TestMySqlClient.java` for it